scylla/src/transport/errors.rs

```diff
@@ -256,6 +256,15 @@ pub enum ProtocolError {
     /// A result with nonfinished paging state received for unpaged query.
     #[error("Unpaged query returned a non-empty paging state! This is a driver-side or server-side bug.")]
     NonfinishedPagingState,
+
+    /// Load balancing policy returned an empty plan - a driver bug!
+    #[error(
+        "Load balancing policy returned an empty plan.\
+        First thing to investigate should be the logic of custom LBP implementation.\
+        If you think that your LBP implementation is correct, or you make use of `DefaultPolicy`,\
+        then this is most probably a driver bug!"
+    )]
+    EmptyPlan,
```

Comment:
How is empty plan related to protocol? I see no reason for it to be in `ProtocolError`.

Reply:
This is violation of a driver's invariant (driver's inner assumption), but it's not necessarily about the protocol (communication with server)

Reply:
True, I've put it in `ProtocolError` without giving it a thought - just because previous version would return `QueryError::ProtocolError(...)`.